Can't build with C++17

Osi relies in 2 files on `std::bind2nd` which has been deprecated in C++11 and removed in C++17: https://en.cppreference.com/w/cpp/utility/functional/bind12

When you compile with C++17 or any more recent standard, obviously it fails:

```
 clang++ -DHAVE_CONFIG_H -I. -I/Users/jenkins/w/prod-v2/BuildSingleReference/p/t/coin-7e1b1ca41dd84/b/src/Osi/src/OsiCommonTest -I../../src/Osi -I/Users/jenkins/w/prod-v2/BuildSingleReference/p/t/coin-7e1b1ca41dd84/b/src/Osi/src/OsiCommonTest/../Osi -I/Users/jenkins/w/prod-v2/BuildSingleReference/p/coin-0d0a7534c642e/p/include -I/Users/jenkins/w/prod-v2/BuildSingleReference/p/coin-0d0a7534c642e/p/include/coin -I/Users/jenkins/w/prod-v2/BuildSingleReference/p/bzip2d4d569d3b1100/p/include -I/Users/jenkins/w/prod-v2/BuildSingleReference/p/zlibe48b509ceab25/p/include -DNDEBUG -stdlib=libc++ -std=gnu++17 -m64 -fPIC -O3 -DOSI_BUILD -MT OsiSolverInterfaceTest.lo -MD -MP -MF .deps/OsiSolverInterfaceTest.Tpo -c /Users/jenkins/w/prod-v2/BuildSingleReference/p/t/coin-7e1b1ca41dd84/b/src/Osi/src/OsiCommonTest/OsiSolverInterfaceTest.cpp -o OsiSolverInterfaceTest.o
/Users/jenkins/w/prod-v2/BuildSingleReference/p/t/coin-7e1b1ca41dd84/b/src/Osi/src/OsiCommonTest/OsiSolverInterfaceTest.cpp:1321:10: error: no member named 'bind2nd' in namespace 'std'
    std::bind2nd(std::plus< double >(), 0.15));
    ~~~~~^
1 error generated.
make[2]: *** [OsiSolverInterfaceTest.lo] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [all-recursive] Error 1
make: *** [all-recursive] Error 1
```

--- a/CHANGELOG
+++ b/CHANGELOG
@@ -1,3 +1,6 @@
+21/03/2023
+  - remove use of std::bind2nd to improve compatibility with C++17, see #177
+
 11/01/2023
   - use new XPRSgetmipentities() function from XPRESS 41
 

--- a/src/OsiCommonTest/OsiSolverInterfaceTest.cpp
+++ b/src/OsiCommonTest/OsiSolverInterfaceTest.cpp
@@ -1321,8 +1321,8 @@ bool test16SebastianNowozin(OsiSolverInterface *si)
   int rows_to_delete_arr[] = { 0 };
   si->deleteRows(1, rows_to_delete_arr);
 
-  std::transform(objective, objective + 4, objective,
-    std::bind2nd(std::plus< double >(), 0.15));
+  for (int i = 0; i < 4; ++i )
+    objective[i] += 0.15;
   si->setObjective(objective);
   si->resolve();
   OSIUNITTEST_ASSERT_ERROR(si->isProvenOptimal(), return false, *si, "test16SebastianNowozin second resolve");

--- a/test/OsiTestSolverInterface.cpp
+++ b/test/OsiTestSolverInterface.cpp
@@ -464,8 +464,8 @@ OsiTestSolverInterface::resolve()
   // Set the dual starting point
   VOL_dvector& dsol = volprob_.dsol;
   dsol.allocate(dsize);
-  std::transform(rowprice_, rowprice_+dsize, dsol.v,
-		 std::bind2nd(std::multiplies<double>(), objsense_));
+  for (i = 0; i < dsize; ++i)
+    dsol.v[i] = rowprice_[i] * objsense_;
 
   // adjust the dual vector (if necessary) to be sure it's feasible
   double * dv = dsol.v;

